### ISSUE TITLE
Refactor CWTBaseTask

### ISSUE THREAD

**Issue by jasonb5** [doutriaux1/wps_cwt#37]
Need to fix CWTBaseTask, some tasks will need to record retries, failures and success while some may not care about successes. i.e. check_auth will want to record retries and failures, while execute will also want to record successes.

Here's two ways to fix it. 
1. Create separate base task classes depending on the events we care about. Since base task classes are only instantiated once and share data, we may loose this ability.
2. Create parameters for the methods denoting what events we care about since they are passed to the on_retry, on_failure, on_success methods.

**Comment by jasonb5** [comment]
Fixed in c535e937850457f811eafacdb8cf4ee69ae27f12, has wrong issue number.

**Comment by jasonb5** [comment]
The solution was to have CWTBaseTask look for an attribute "PUBLISH", which is a binary flag. The module wps.processes.process as SUCCESS, FAILURE, RETRY and ALL defined. You can assign any combination of these to PUBLISH, see example below. This solution was chosen because task shared a singleton base class so the flag must be declared on an instance of the task.

```python
from wps.processes import process

@process.cwt_shared_task
def test(self):
  self.PUBLISH = process.SUCCESS | process.RETRY | process.FAILURE # equivalent process.ALL
```
The CWTBaseTask methods on_sucess, on_failure and on_retry can all be called.

```python
@process.cwt_shared_task
def test(self):
  self.PUBLISH = process.RETRY | process.FAILURE
```

This will only allow on_failure and on_retry to be called. If the process succeeds on_sucess will not be called.
